Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -423,9 +437,10 @@ class DropDownMenu
       if (DropDownMenu._repositionTimer)
         clearTimeout (DropDownMenu._repositionTimer);
 
-      if (DropDownMenu._currentPopup && DropDownMenu._currentPopup == div && $(div).is (':visible'))
+      const isDivVisible = (div.offsetWidth > 0 || div.offsetHeight > 0);
```

Comment:
@a-ctor the intention of this code was to test via CSS / layout information, not height/width. I did not do a deep dive, are offsetWidth/Height 0 when the element is set to display:none ?

Reply:
Yes they are set to `0`.

Reply:
@a-ctor since it popped up here and there've been a couple other more complex steps in this file or one of the others (e.g. applying of a dictioanry of attributes): should we introduce utilities for these type of logic blocks? They do appear to be sufficiently complicated to "get right".

Reply:
Removing code duplication is probably a good idea. The visibility check could be put into `LayoutHelper.IsVisible(element)`. For other utilities we would need to create a new class to make the name fit.

Reply:
The best solution would probably be to create the utility methods as soon as the need arises, but in a separate PR. It's also okay if we start with TODO comments and then update the code later on. The third option would be to create them in the branch as separate commits and just resolve merge conflicts in the utility when they arise. We should aim to prioritize de-blocking and actively track code duplications where they cannot be avoided.

Reply:
So every utility method gets a separate ticket, fixed in a separate PR? Then I'll add a ticket for `LayoutHelper.IsVisible(element)` and insert a todo here.